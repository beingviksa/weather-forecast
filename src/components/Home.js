import FamousPlaces from "./FamousPlaces";
import SearchBox from "./SearchBox";

const Home = () => {
  return (
    <div className="container">
      <SearchBox />
      <FamousPlaces />
    </div>
  );
};

export default Home;
