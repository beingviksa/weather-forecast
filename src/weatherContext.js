import { createContext, useContext, useState } from "react";

export const WeatherContext = createContext();

export const WeatherContextProvider = ({ children }) => {
  const [searchResults, setSearchResults] = useState([]);
  const [consolidatedData, setConsolidatedData] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState("");
  return (
    <WeatherContext.Provider
      value={{
        searchResults,
        setSearchResults,
        consolidatedData,
        setConsolidatedData,
        isLoading,
        setIsLoading,
        error,
        setError,
      }}>
      {children}
    </WeatherContext.Provider>
  );
};

export const useWeatherContext = () => {
  return useContext(WeatherContext);
};
