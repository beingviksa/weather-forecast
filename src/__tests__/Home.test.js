import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Home from "../components/Home";
import { WeatherContext } from "../weatherContext";

describe("<Home/>", () => {
  const contextData = {
    searchResults: [
      [
        {
          id: 2643734,
          name: "Londonderry County Borough",
          state: "",
          country: "GB",
          coord: {
            lon: -7.30917,
            lat: 54.997211,
          },
          slug: "londonderry-county-borough-2643734",
        },
        {
          id: 2643743,
          name: "London",
          state: "",
          country: "GB",
          coord: {
            lon: -0.12574,
            lat: 51.50853,
          },
          slug: "london-2643743",
        },
      ],
    ],
    setSearchResults: jest.fn(),
  };
  it("render the Home Page", async () => {
    const { container } = render(
      <WeatherContext.Provider value={contextData}>
        <Router>
          <Home />
        </Router>
      </WeatherContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("render search box and famous places on home page", async () => {
    const { getByTestId } = render(
      <WeatherContext.Provider value={contextData}>
        <Router>
          <Home />
        </Router>
      </WeatherContext.Provider>
    );

    const searchBox = getByTestId("search-box");
    expect(searchBox).toBeInTheDocument();

    const famousPlaces = getByTestId("famous-places");
    expect(famousPlaces).toBeInTheDocument();
  });
});
