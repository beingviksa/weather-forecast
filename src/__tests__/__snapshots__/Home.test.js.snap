// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> render the Home Page 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="search"
    >
      <input
        data-testid="search-box"
        placeholder="Seach for a city..."
        type="text"
        value=""
      />
    </div>
    <div
      class="places"
      data-testid="famous-places"
    >
      <div
        class="places__row"
      >
        <div
          class="places__box"
        >
          <a
            href="/location/london-2643743"
          >
            <div
              class="places__image-wrapper"
            >
              <img
                alt="London"
                src="london.jpg"
              />
            </div>
            <span>
              London
            </span>
          </a>
        </div>
        <div
          class="places__box"
        >
          <a
            href="/location/paris-2968815"
          >
            <div
              class="places__image-wrapper"
            >
              <img
                alt="Paris"
                src="paris.jpg"
              />
            </div>
            <span>
              Paris
            </span>
          </a>
        </div>
        <div
          class="places__box"
        >
          <a
            href="/location/tokyo-1850147"
          >
            <div
              class="places__image-wrapper"
            >
              <img
                alt="Tokyo"
                src="tokyo.jpg"
              />
            </div>
            <span>
              Tokyo
            </span>
          </a>
        </div>
        <div
          class="places__box"
        >
          <a
            href="/location/new-york-city-5128581"
          >
            <div
              class="places__image-wrapper"
            >
              <img
                alt="New York"
                src="new-york.jpg"
              />
            </div>
            <span>
              New York
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
